src/worker.cpp

```diff
@@ -1547,4 +1547,96 @@ void Worker::performBackupAndRestore(types::VPDMapVariant& io_srcVpdMap)
         */
     }
 }
+
+void Worker::deleteFruVpd(const std::string& i_dbusObjPath)
+{
+    if (i_dbusObjPath.empty())
+    {
+        throw std::runtime_error(
+            "Given input path is empty. Cannot perform delete FRU VPD for: " +
+            i_dbusObjPath);
+    }
+
+    try
+    {
+        const bool l_presentPropValue =
+            std::get<bool>(dbusUtility::readDbusProperty(
+                constants::pimServiceName, i_dbusObjPath,
+                constants::inventoryItemInf, "Present"));
+
+        if (!l_presentPropValue)
+        {
+            logging::logMessage(
+                "Given FRU is not present. Could not perform delete FRU VPD for " +
+                i_dbusObjPath);
+            return;
+        }
+        else
+        {
+            if (!m_parsedJson.empty() &&
+                !(jsonUtility::getInventoryObjPathFromJson(m_parsedJson,
```

Comment:
I guess only isActionRequired() API is enough, no need to have this check.
we can add 'm_parsedJson' empty check in the later condition itself. 
Please verify this.

Reply:
Done